src/NextGen/Framework/FormTemplates/ServiceProvider.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+namespace Give\NextGen\Framework\FormTemplates;
+
+use Give\Helpers\Hooks;
+use Give\NextGen\Framework\FormTemplates\Actions\RegisterFormTemplates;
+use Give\NextGen\Framework\FormTemplates\Registrars\FormTemplateRegistrar;
+use Give\ServiceProviders\ServiceProvider as ServiceProviderInterface;
+
+/**
+ * @unreleased
+ */
+class ServiceProvider implements ServiceProviderInterface
+{
+    /**
+     * @inheritDoc
+     */
+    public function register()
+    {
+        give()->singleton(FormTemplateRegistrar::class);
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function boot()
+    {
+        $formTemplateRegistrar = give(FormTemplateRegistrar::class);
+
+        Hooks::doAction('givewp_register_form_template', $formTemplateRegistrar);
```

Comment:
Side note: Maybe we should consider overloading the signature for the `Hooks` class methods, so we could pass multiple actions and avoid intermediary variables. Just a thought, but that would be in core.